packages/accounts-controller/src/AccountsController.ts

```diff
@@ -0,0 +1,580 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+import { SnapKeyring } from '@metamask/eth-snap-keyring';
+import type { InternalAccount } from '@metamask/keyring-api';
+import { EthAccountType } from '@metamask/keyring-api';
+import type {
+  KeyringControllerState,
+  KeyringController,
+  KeyringControllerEvents,
+} from '@metamask/keyring-controller';
+import type {
+  SnapControllerEvents,
+  SnapControllerState,
+} from '@metamask/snaps-controllers';
+import { sha256FromString } from 'ethereumjs-util';
+import type { Patch } from 'immer';
+import { v4 as uuid } from 'uuid';
+
+const controllerName = 'AccountsController';
+
+export type AccountsControllerState = {
+  internalAccounts: {
+    accounts: Record<string, InternalAccount>;
+    selectedAccount: string; // id of the selected account
+  };
+};
+
+export type AccountsControllerGetStateAction = {
+  type: `${typeof controllerName}:getState`;
+  handler: () => AccountsControllerState;
+};
+
+export type AccountsControllerActions = AccountsControllerGetStateAction;
```

Comment:
For the same reason, would be nice to provide access to `AccountsController`'s methods also through the messenger: This would allow other `BaseControllerV2`s to consume its methods through its messenger, instead of passing the entire (or part) controller instance to the constructors